Add CSV

secretGeek opened the pull request:
A growing list of powerful tools for manipulating and validating CSV, the format that data-scientists eat for breakfast.

<!-- Congrats on creating an Awesome list! 🎉 -->


<!-- Please fill in the below placeholders -->

**https://github.com/secretGeek/AwesomeCSV**

**CSV: there are many awesome tools for utlizing the most common format of data-scientists.**


### By submitting this pull request I confirm I've read and complied with the below requirements 🖖

**Please read it multiple times. I spent a lot of time on these guidelines and most people miss a lot.**

## Requirements for your pull request

- **You have to review at least 2 other [open pull requests](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen).** Try to prioritize unreviewed PRs, but you can also add more comments to reviewed PRs. Go through the below list when reviewing. This requirement is meant to help make the Awesome project self-sustaining. Comment here which PRs you reviewed. You're expected to put a good effort into this and to be thorough. Look at previous PR reviews for inspiration.

I read through a lot of PR's and added comments at:
#1446 
#1457 

- I have read and understood the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/master/create-list.md).
- This pull request has a descriptive title in the format `Add Name of List` (Example: `Add Swift`), not `Update readme.md` or `Add awesome list`.
- The entry in the Awesome list should:
	- Include a short description about the project/theme of the list. **It should not describe the list itself.**<br>Example: `- [Fish](…) - User-friendly shell.`, not `- [Fish](…) - Resources for Fish.`.
	- Be added at the bottom of the appropriate category.
- The list I'm submitting complies with the below requirements.


## Requirements for your Awesome list

- **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
- It's the result of hard work and the best I could possibly produce.
- Non-generated Markdown file in a GitHub repo.
- **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
- The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
- Not a duplicate.
- Only has awesome items. Awesome lists are curations of the best, not everything.
- Includes a project logo/illustration whenever possible.
	- Either centered, fullwidth, or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
	- The image should link to the project website or any relevant website.
	- The image should be high-DPI. Set it to maximum half the width of the original image.
- Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
- Includes the [Awesome badge](https://github.com/sindresorhus/awesome/blob/master/awesome.md#awesome-badge).
	- Should be placed on the right side of the readme heading.
		- Can be placed centered if the list has a centered graphics header.
	- Should link back to this list.
- Has a Table of Contents section.
	- Should be named `Contents`, not `Table of Contents`.
	- Should be the first section in the list.
	- Should only have one level of [nested lists](https://commonmark.org/help/tutorial/10-nestedLists.html), preferably none.
- Has an [appropriate license](https://github.com/sindresorhus/awesome/blob/master/awesome.md#choose-an-appropriate-license).
	- That means something like CC0, **not a code licence like MIT, BSD, Apache, etc.**
	- [WTFPL](http://www.wtfpl.net) and [Unlicense](http://unlicense.org) are not acceptable licenses.
	- If you use a license badge, it should be SVG, not PNG.
- Has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/awesome.md#include-contribution-guidelines).
	- The file should be named `contributing.md`. Casing is up to you.
- Has consistent formatting and proper spelling/grammar.
	- The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
	- The description starts with an uppercase character and ends with a period.
	- Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
- Doesn't include a Travis badge.<br>You can still use Travis for list linting, but the badge has no value in the readme.

Go to the top and read it again.

secretGeek commented:
(I ran awesome-lint on AwesomeCSV and fixed up everything.)

(The encouragement to review other PR's is a great idea. It gave me much more insight and empathy into the awesome-list world and the challenges of being a maintainer.)

secretGeek commented:
@sfischer13 Thanks for looking at this!
When you dig into those instances, I think they're deficencies in awesome-lint.

Items where i've included asterisks for footnotes confuse the tool. Similarly, having 'footnotes' last confused the tool as it insisted that License had to be last (to which I acquiesced). Basically it can't handle the concept of footnotes (and why should it I guess). All I'm saying is, don't blindly trust a tool. 

Thanks again for looking. May all your embedded delimiters be qualified!

sindresorhus commented:
The section titles should not contain the word `Awesome`. That's moot.

sindresorhus commented:
You need to actually add your email to this: https://github.com/secretGeek/AwesomeCSV/blob/master/code-of-conduct.md

sindresorhus commented:
> * **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)

sindresorhus commented:
```
❯ awesome-lint https://github.com/secretGeek/AwesomeCSV
✖ Linting

  README.md:41:3
  ✖  32:49  List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖  41:3   Link to http://www.wise.io/tech/paratext is dead              remark-lint:no-dead-urls
  ✖  74:55  List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖  83:77  List item link and description must be separated with a dash  remark-lint:awesome/list-item

  4 errors
```

https://github.com/sindresorhus/awesome-lint

sindresorhus commented:
> * The description **starts with an uppercase character** and ends with a period.

sindresorhus commented:
This `META: Other Lists of CSV Tools and References` needs a better and shorter title.

sindresorhus commented:
Generally improve the list and its formatting.

sindresorhus commented on readme.md:
```diff
@@ -506,6 +506,7 @@
 - [JSON](https://github.com/burningtree/awesome-json#readme) - Text based data interchange format.
 	- [GeoJSON](https://github.com/tmcw/awesome-geojson#readme)
 	- [Datasets](https://github.com/jdorfman/awesome-json-datasets#readme)
+- [CSV](https://github.com/secretGeek/awesomeCSV#readme) - The most futuristic data format from the distant past.
```
The description should describe what it is. This sounds more like a marketing tagline.

secretGeek replied:
Added new commit which updates this description to "Comma-Separated-Text, an ancient format that remains widely used."

sindresorhus replied:
That still doesn't describe **what** it is... 

You could have just copy-pasted something from Wikipedia: https://en.wikipedia.org/wiki/Comma-separated_values

I would recommend something like this:

> A text file format that stores tabular data and uses a comma to separate values.

secretGeek replied:
I went with exactly that, thank you.

secretGeek commented:
Responses for @sindresorhus 

Thanks for taking the time to review the list so thoroughly.

> Includes a succinct description of the project/theme at the top of the readme.

Done. ("A carefully curated list of CSV-related tools and resources.") (I added it as a sub-heading, as this seemed reasonable.

> 41:3   Link to http://www.wise.io/tech/paratext is dead              remark-lint:no-dead-urls

Fixed. (Pointed to source code of the related project.)

>  32:49  List item link and description must be separated with a dash  remark-lint:awesome/list-item
>  74:55  List item link and description must be separated with a dash  remark-lint:awesome/list-item
>  83:77  List item link and description must be separated with a dash  remark-lint:awesome/list-item

...these are due to the footnote markers, see https://github.com/sindresorhus/awesome/pull/1461#issuecomment-446800799 , the list item link and description *are* separated with a dash.

> The section titles should not contain the word `Awesome`. That's moot.

Done.

> You need to actually add your email to this: https://github.com/secretGeek/AwesomeCSV/blob/master/code-of-conduct.md

Done. Thank you for noticing.

> The description **starts with an uppercase character** and ends with a period.

Unclear if this is a comment or a request.

> This `META: Other Lists of CSV Tools and References` needs a better and shorter title.

Done.

> Generally improve the list and its formatting.

I've done what I can. 

> The description should describe what it is. This sounds more like a marketing tagline.

I've added a fresh commit to my pr, hopefully it's more suitable.

kind regards
lb

sindresorhus commented:
> Unclear if this is a comment or a request.

Request

secretGeek commented:
Ready for further review. All issues addressed.

ehmicky commented:
Code reviewed it as well. Looks all good now.

Only thing I could find: a CSV logo might be a good idea.

sindresorhus commented:
`## A carefully curated list of CSV-related tools and resources` => `> A carefully curated list of CSV-related tools and resources`

---

Linkify `CSV` in `CSV remains the most futuristic data format from the distant past.` to https://en.wikipedia.org/wiki/Comma-separated_values

sindresorhus commented:
```
❯ awesome-lint https://github.com/secretGeek/AwesomeCSV
✖ Linting

  README.md:46:3
  ✖   1:1   License was not detected by GitHub                                                                     remark-lint:awesome/github
  ✖  12:1   Table of Contents must be the first section                                                            remark-lint:awesome/toc
  ✖  23:1   ToC item "META: Other similar lists" does not match corresponding heading "Meta: Other similar lists"  remark-lint:awesome/toc
  ✖  34:49  List item link and description must be separated with a dash                                           remark-lint:awesome/list-item
  ✖  46:3   Link to http://streamingpivot.com/ is dead                                                             remark-lint:no-dead-urls
  ✖  62:3   Link to http://www.csvstudio.com is dead                                                               remark-lint:no-dead-urls
  ✖  69:27  List item description must end with proper punctuation                                                 remark-lint:awesome/list-item
  ✖  78:55  List item link and description must be separated with a dash                                           remark-lint:awesome/list-item
  ✖  87:77  List item link and description must be separated with a dash                                           remark-lint:awesome/list-item

  9 errors
```

https://github.com/sindresorhus/awesome-lint

secretGeek commented:
Have gone through and made updates based on all feedback to date. Thank you @sindresorhus

sindresorhus commented:
Tweet: https://twitter.com/awesome__re/status/1133813876242374656
